Flyout menu implementation

msl5994 opened the pull request:
Created the Flyout shell
Added shell navigation
MVVM for the QuestionAskerPage
App.ReadingContext global variable

msl5994 committed:
Added Flyout menu with underlying Shell Navigation.
Replaced _readingContext variable, with a static App global variable.

msl5994 committed:
Started adding a ViewModel for Questions. Updated to XamarinForms 5.0 and added XCT nuget package.

msl5994 committed:
Continued moving methods into the ViewModel  from the code behind for consolidation and use in button Commands (with parameters) and added a convereter for booleans for data binding with IsVisible.

msl5994 committed:
QuestionAskerPage is MVVM completed at the moment.

msl5994 committed:
Finished implementing MVVM for the QuestionAskerPage. Fixed some of the navigations. Debugged the Httpclient timeout.

msl5994 committed:
Finished creating the binding for MPButtonsEnabled to update in real time after the data is initialized from either the server, or the JSON.

msl5994 committed:
MVVM for SecondPage has been implemented. Added the text for MP Data still initializing.
Removed my global variable for MPDataInitialized because it was unnecessary, as the ParliamentData class is static.

vteague commented:
Hi @msl5994,
A couple of small glitches I've noticed so far:

-    When I enter a question it doesn't seem to be reflected, either in the "draft question" view at the bottom of the readingpage, or in the questionDetailPage at the end of the flow.

- With the server off, I choose an authority or MP to answer the question, "not sure" to raise the question, then when I get to RegisterPage1 and am trying to set up an account (in preparation for uploading my question), pushing the back button crashes the app.

- When I draft a question and choose 'My MP', the MP I chose shows up in the Filters on the ReadingPage under "other MPs who should answer the question". It should be "My MPs who should answer the question"

- Trying to add a link or answer in the QuestionDetailPage crashed the app (I'm not sure you introduced this one - possibly it's always been there).

vteague commented:
- It'd be good to put the server public key in a file and add it to .gitignore (even though not really part of the flyout menu.)

vteague commented:
Now with the server on:
- When I got to the 'create account' page without having identified my electorates, and clicked on "next: find your electorates" it crashed.
- Similarly, when I got to the 'create account' page after identifying my MPs (because I'd asked for something to be raise by my MP in an earlier frame), and clicked on 'complete account creation' it also crashed.

I get an unhandled exception at Line 109 of QuestionDetailPage:
System.NullReferenceException: Object reference not set to an instance of an object.
  at RightToAskClient.Views.QuestionDetailPage.setSuggester (System.Object sender, System.EventArgs e) [0x00001] in /home/vanessa/Documents/RightToAsk/RightToAskOrg/righttoask-app/RightToAskClient/RightToAskClient/RightToAskClient/Views/QuestionDetailPage.xaml.cs:101 
  at Xamarin.Forms.Page.SendDisappearing () [0x00046] in D:\a\1\s\Xamarin.Forms.Core\Page.cs:476 
  at Xamarin.Forms.ShellSection.PresentedPageDisappearing () [0x00016] in D:\a\1\s\Xamarin.Forms.Core\Shell\ShellSection.cs:925 
  at Xamarin.Forms.ShellSection.OnPushAsync (Xamarin.Forms.Page page, System.Boolean animated) [0x00046] in D:\a\1\s\Xamarin.Forms.Core\Shell\ShellSection.cs:838 
  at Xamarin.Forms.ShellSection.PushStackOfPages (System.Collections.Generic.List`1[T] pages, System.Nullable`1[T] animate) [0x00077] in D:\a\1\s\Xamarin.Forms.Core\Shell\ShellSection.cs:600

vteague commented:
I also get a compiler warning at line 113 of QuestionDetailPage.xaml.cs:
over App.ReadingContext.DraftQuestion = null;    
I think just set it to "";
This will let the null-checkers do their work. (Again, I don't think you necessarily introduced this.)

msl5994 committed:
Removed ReadingPage from FlyoutMenu.
Fixed DraftQuestion data bind that I had messed with.

msl5994 committed:
Added a file for PublicServerKey
the gitIgnore ignores that file
Signature service reads SPKI from that file

Fixed a few crashes due to navigation or some properties not being initialized due to newer data binds.

msl5994 committed:
Fixed the button Commands for Answering MPs.
Started cleaning up the Warnings (reduced from 100+ down to 40)

msl5994 committed:
Testing a fix for SPIKRaw base64 string decoding error.
Returned the naming convention of a few variables to their original state.

msl5994 committed:
Changed Console.WriteLine to Debug.WriteLine.
Caught an exception for decoding the PublicServerKey.

vteague commented:
Terrific. Thanks. Looks great.
